subprojects/packagefiles/fluidsynth/meson.build

```diff
@@ -72,6 +94,21 @@ gthread_dep = dependency('gthread-2.0', static : get_option('try-static-deps'))
 openmp_dep  = dependency('openmp', required : false, language : 'c',
                          version : '>= 4.0')
 
+ext_deps = [m_dep, glib_dep, gthread_dep, openmp_dep]
+
+# Windows specific options
+lib_args = []
+if host_machine.system() == 'windows'
+  if is_shared_lib
+    lib_args = ['-DFLUIDSYNTH_DLL_EXPORTS']
+  endif
+  # Specific workaround on MSYS2 to obtain static dependencies of intl, such as iconv
+  # This is required as of meson 0.59
+  if get_option('try-static-deps') and cc.get_id() in ['gcc', 'clang'] 
+    ext_deps += [dependency('intl' , static : true)]
```

Comment:
Is there any problem of doing this check without bumping the minimum required Meson version?

The only platform that requires this (MSYS2) has Meson 0.59 available.

Reply:
The platforms which don't require it, will still fail to "find" it, and error out.

If you use `required: false` then on versions of meson less than 0.60 it will just fail to find it at all. You may want to detect the specific case of MSYS2 though, and in that case set it to `required: true`.

Reply:
The fact that meson doesn't allow getting environment variables is a pain. That is by far the easiest way to detect MSYS2.

I may have to detect if `sh` is available, and if so, check with a shell one-liner.

Reply:
Added a proper MSYS2 check. This branch should only be run when host is Windows and MSYS2 is detected.

Reply:
you can check with `python -c 'import os; print(os.getenv("MSYSTEM", ""))'` and python is more likely installed than sh on Windows... and guaranteed if you use meson, less guaranteed if you use https://sr.ht/~lattis/muon but whatever, it still works as long as python is installed.